### PR TITLE
Enhancement Hardcoded api endpoints 486

### LINKED ISSUE #486: Hardcoded api endpoints (Admin Panel)
Currently, all of the API endpoints that _AdminPanel_ services make requests to are hardcoded.

For example here
https://github.com/ita-social-projects/WhatBackend/blob/0303e1f35b854ef3646fbb3243dfa85704fe75e3/CharlieBackend.AdminPanel/Services/CourseService.cs#L36
or here
https://github.com/ita-social-projects/WhatBackend/blob/0303e1f35b854ef3646fbb3243dfa85704fe75e3/CharlieBackend.AdminPanel/Services/StudentService.cs#L43

Possible solution:
Move these hardcoded strings into _appsettings.json_ and read them along the other application settings.

### PR DESCRIPTION
**Pull request by Vso1o**
#486 Move hardcoded strings into appsettings.json and read them along the other application settings.

### CODE REVIEW

File: CharlieBackend.AdminPanel/ApplicationSettings.cs [hunk 1]
```diff
@@ -39,6 +46,59 @@ public class CoursesApiEndpoints
...
+    public class StudentsApiEndpoints
+    {
+        public string ActiveStudentEndpoint { get; set; }
+        public string GetAllStudentsEndpoint { get; set; }
+        public string AddStudentEndpoint { get; set; }
+        public string GetStudentEndpoint { get; set; }
+        public string UpdateStudentEndpoint { get; set; }
+        public string DisableStudentEndpoint { get; set; }
+        public string EnableStudentEndpoint { get; set; }
+
```
Comment (andrii-developer) [inline CharlieBackend.AdminPanel/ApplicationSettings.cs]: Please remove these empty lines before closed brackets.

File: CharlieBackend.AdminPanel/Services/CourseService.cs [hunk 2]
```diff
@@ -17,6 +17,7 @@ public class CourseService : ICourseService
 
         private readonly CoursesApiEndpoints _coursesApiEndpoints;
 
+
```
Comment (andrii-developer) [inline CharlieBackend.AdminPanel/Services/CourseService.cs]: Please remove this empty line.

File: CharlieBackend.AdminPanel/Services/MentorService.cs [hunk 3]
```diff
@@ -18,34 +18,62 @@ public class MentorService : IMentorService
...
         }
 
         public async Task<IList<MentorViewModel>> GetAllMentorsAsync()
         {
+            var getAllMentorsEndpoint = string
+                .Format(_mentorsApiEndpoints.GetAllMentorsEndpoint);
+
+            var activeMentorEndpoint = string
+                .Format(_mentorsApiEndpoints.ActiveMentorEndpoint);
+
```
Comment (andrii-developer) [inline CharlieBackend.AdminPanel/Services/MentorService.cs]: Please remove this empty line.

File: CharlieBackend.AdminPanel/Services/StudentGroupService.cs [hunk 4]
```diff
@@ -23,24 +23,32 @@ public class StudentGroupService : IStudentGroupService
...
+                                   ICourseService courseService,
+                                   IOptions<ApplicationSettings> options)
         {
             _apiUtil = apiUtil;
             _mapper = mapper;
 
             _mentorService = mentorService;
             _studentService = studentService;
             _courseService = courseService;
+
```
Comment (andrii-developer) [inline CharlieBackend.AdminPanel/Services/StudentGroupService.cs]: Please remove this empty line.

File: CharlieBackend.AdminPanel/Services/StudentService.cs [hunk 5]
```diff
@@ -17,15 +17,29 @@ public class StudentService: IStudentService
     {
         private readonly IApiUtil _apiUtil;
 
-        public StudentService(IApiUtil apiUtil)
+        private readonly StudentsApiEndpoints _studentsApiEndpoints;
+
+        private readonly StudentGroupsApiEndpoints _studentGroupsApiEndpoints;
+
```
Comment (andrii-developer) [inline CharlieBackend.AdminPanel/Services/StudentService.cs:21]: Please remove this empty line.

**Comment by gingerbread-monster** [comment]
Please don't forget to link you pull request to a corresponding issue next time.

File: CharlieBackend.AdminPanel/Services/MentorService.cs [hunk 6]
```diff
@@ -18,34 +18,62 @@ public class MentorService : IMentorService
...
+            return await _apiUtil.EnableAsync<bool>(enableMentorEndpoint);
         }
 
         public async Task<IList<MentorViewModel>> GetAllMentorsAsync()
         {
+            var getAllMentorsEndpoint = string
+                .Format(_mentorsApiEndpoints.GetAllMentorsEndpoint);
+
+            var activeMentorEndpoint = string
+                .Format(_mentorsApiEndpoints.ActiveMentorEndpoint);
```
Comment (gingerbread-monster) [inline CharlieBackend.AdminPanel/Services/MentorService.cs]: Why do you use `string.Format()` here? 😄

File: CharlieBackend.AdminPanel/Services/CalendarService.cs [hunk 7]
```diff
@@ -23,22 +24,29 @@ public class CalendarService : ICalendarService
...
-        private const string _getActiveCoursesEndpoint = "api/courses/isActive";
-        private const string _getActiveMetorsEndpoint = "api/mentors/active";
-        private const string _getStudentGroupsEndpoint = "api/student_groups";
-        private const string _getActiveStudentsEndpoint = "api/students/active";
-        private const string _getThemesEndpoint = "api/themes";
+        private readonly string _getActiveCoursesEndpoint;// = "api/courses/isActive";
+        private readonly string _getActiveMetorsEndpoint;// = "api/mentors/active";
+        private readonly string _getStudentGroupsEndpoint;// = "api/student_groups";
+        private readonly string _getActiveStudentsEndpoint;// = "api/students/active";
+        private readonly string _getThemesEndpoint;// = "api/themes";
```
Comment (gingerbread-monster) [inline CharlieBackend.AdminPanel/Services/CalendarService.cs]: Remove the comments please. They can only mislead, not help.

File: CharlieBackend.AdminPanel/Services/StudentGroupService.cs [hunk 8]
```diff
@@ -23,24 +23,30 @@ public class StudentGroupService : IStudentGroupService
         private readonly IStudentService _studentService;
         private readonly ICourseService _courseService;
```
Comment (andrii-developer) [inline CharlieBackend.AdminPanel/Services/StudentGroupService.cs]: Please remove that blank line.

File: CharlieBackend.AdminPanel/Services/StudentService.cs [hunk 9]
```diff
@@ -17,15 +17,26 @@ public class StudentService: IStudentService
     {
         private readonly IApiUtil _apiUtil;
 
-        public StudentService(IApiUtil apiUtil)
+        private readonly StudentsApiEndpoints _studentsApiEndpoints;
+
```
Comment (andrii-developer) [inline CharlieBackend.AdminPanel/Services/StudentService.cs]: Please remove blank lines between private properties.

File: CharlieBackend.AdminPanel/Services/StudentService.cs [hunk 10]
```diff
@@ -17,15 +17,26 @@ public class StudentService: IStudentService
...
+        private readonly StudentGroupsApiEndpoints _studentGroupsApiEndpoints;
+
+        public StudentService(IApiUtil apiUtil, IOptions<ApplicationSettings> options)
         {
             _apiUtil = apiUtil;
+
```
Comment (andrii-developer) [inline CharlieBackend.AdminPanel/Services/StudentService.cs]: Please remove these blank lines in the constructor.

File: CharlieBackend.AdminPanel/Services/ThemeService.cs [hunk 11]
```diff
@@ -14,33 +14,47 @@ public class ThemeService : IThemeService
     {
         private readonly IApiUtil _apiUtil;
 
-        public ThemeService(IApiUtil apiUtil)
+        private readonly ThemesApiEndpoints _themesApiEndpoints;
+
```
Comment (andrii-developer) [inline CharlieBackend.AdminPanel/Services/ThemeService.cs:17]: Please, remove that blank line.

File: CharlieBackend.AdminPanel/Services/ScheduleService.cs [hunk 12]
```diff
@@ -14,15 +15,19 @@ class ScheduleService : IScheduleService
     {
         private readonly IApiUtil _apiUtil;
```
Comment (andrii-developer) [inline CharlieBackend.AdminPanel/Services/ScheduleService.cs]: Please, remove that blank line.

File: CharlieBackend.AdminPanel/Services/MentorService.cs [hunk 13]
```diff
@@ -18,34 +18,58 @@ public class MentorService : IMentorService
     {
         private readonly IApiUtil _apiUtil;
```
Comment (andrii-developer) [inline CharlieBackend.AdminPanel/Services/MentorService.cs]: Please, remove blank lines between private fields and in the constructor.

File: CharlieBackend.AdminPanel/Services/StudentGroupService.cs [hunk 14]
```diff
@@ -18,29 +18,32 @@ public class StudentGroupService : IStudentGroupService
...
         private readonly IApiUtil _apiUtil;
         private readonly IMapper _mapper;
-
         private readonly IMentorService _mentorService;
         private readonly IStudentService _studentService;
         private readonly ICourseService _courseService;
+        private readonly StudentGroupsApiEndpoints _studentGroupsApiEndpoints;
 
         public StudentGroupService(IApiUtil apiUtil,
                                    IMapper mapper,
```
Comment (andrii-developer) [inline CharlieBackend.AdminPanel/Services/StudentGroupService.cs]: Please remove this empty line.

**Review by andrii-developer — APPROVED** [review]

**Review by Pasha1968 — APPROVED** [review]

**Review by DemetriosKoras — APPROVED** [review]

**Pull request merged**